crates/debugging/src/trace.rs

```diff
@@ -27,24 +29,39 @@ pub struct TraceInfo {
     pub call_result: CallResult,
 }
 
+#[derive(Debug, Clone)]
+pub struct ContractName(pub String);
+
 #[derive(Debug, Clone)]
 pub struct StorageAddress(pub ContractAddress);
 
 #[derive(Debug, Clone)]
 pub struct CallerAddress(pub ContractAddress);
 
 impl Trace {
-    /// Creates a new [`Trace`] from a given `cheatnet` [`CallTrace`].
-    pub fn from_call_trace(call_trace: &Rc<RefCell<CallTrace>>) -> Self {
+    /// Creates a new [`Trace`] from a given `cheatnet` [`CallTrace`] and [`ContractsData`].
+    pub fn from_call_trace(
+        call_trace: &Rc<RefCell<CallTrace>>,
+        contracts_data: &ContractsData,
+    ) -> Self {
         let call_trace = call_trace.borrow();
         let nested_calls = call_trace
             .nested_calls
             .iter()
             .filter_map(CallTraceNode::extract_entry_point_call)
-            .map(Self::from_call_trace)
+            .map(|call_trace| Self::from_call_trace(call_trace, contracts_data))
             .collect();
 
         let trace_info = TraceInfo {
+            contract_name: contracts_data
+                .get_contract_name(
+                    &call_trace
+                        .entry_point
+                        .class_hash
+                        .expect("this should be set in `fn execute_call_entry_point` in cheatnet"),
+                )
+                .cloned()
+                .map(ContractName),
```

Comment:
nit: Let's move `contract_name` declaration before `trace_info`, I'm not a fan of these deep nestings 😅

Reply:
Maybe extract this to a helper function? This clutters this code too much.

Reply:
Can be too